tests/mock_network/utils.rs

```diff
@@ -743,38 +802,137 @@ pub fn verify_invariants_for_nodes(env: &Environment, nodes: &[TestNode]) {
     }
 }
 
+// Send an `UserMessage` with `content` from `src` to `dst`.
+pub fn send_user_message(
+    nodes: &mut [TestNode],
+    src: Prefix<XorName>,
+    dst: Prefix<XorName>,
+    content: Vec<u8>,
+) {
+    let src_location = SrcLocation::Section(src);
+    let dst_location = DstLocation::Section(dst.name());
+
+    trace!(
+        "send_user_message: {:?} -> {:?}: {:?}",
+        src_location,
+        dst_location,
+        hex_fmt::HexFmt(&content),
+    );
+
+    for node in elders_with_prefix_mut(nodes, &src) {
+        node.inner
+            .send_message(src_location, dst_location, content.clone())
+            .unwrap()
+    }
+}
+
+// Poll until all sections have up-to-date knowledge of their neighbour sections.
+// We consider section A's knowledge of its neighbour section B as up-to-date if every elder from A
+// knows at least `threshold` online nodes from B.
+pub fn update_neighbours_and_poll(env: &Environment, nodes: &mut [TestNode], threshold: usize) {
+    info!("update_neighbours_and_poll start");
+
+    let mut rng = env.new_rng();
+
+    // Max number of times we will try to send messages and wait until the knowledge gets updated.
+    let max_sends = 3;
```

Comment:
Is it not guaranteed that one message will be enough to trigger the update?

Reply:
Yes, it can fail for at least these reasons:
1) the `UserMessage` might fail to accumulate at the source if the section is undergoing churn
2) the `SendNeighbourInfo` and/or `NeighbourInfo` votes might fail to accumulate if the section doing the voting is undergoing split

We could solve 1 by synchronizing the send via parsec and 2 by resending the message to the sibling and re-voting, which is roughly what we were doing previously. We needed to do that because the `NeighbourInfo` message was sent only once and so we needed to make sure it's delivered. That was a flawed approach, because the delivery can never be 100% guaranteed. Now the message is sent every time we detect the knowledge is out of date and so it isn't that critical that its delivery succeeds. This allows as to remove some complexity.

Reply:
Ah, ok. Thanks for the clarification.